### ISSUE TITLE
MLKInvalidLocalModel on IOS

### ISSUE THREAD

**Issue by dimitribocquet** [bharat-biradar/Google-Ml-Kit-plugin#179]
I am trying to use `imageLabeler` with a custom model, but I always have this error:

```
The selected imageFormatGroup is not supported by iOS. Defaulting to brga8888
*** Terminating app due to uncaught exception 'MLKInvalidLocalModel', reason: 'Local model path (model.tflite) is invalid.'
*** First throw call stack:
(0x181f1dcac 0x198f8c758 0x181f70d14 0x105bc44dc 0x105bc4044 0x108a22608 0x108a21dc8 0x108a21b30 0x108a2193c 0x10e32164c 0x10de0eac4 0x10de14708 0x10e1f93c0 0x10e105510 0x10e10bfc0 0x181f31fdc 0x181ebc348 0x181eb6fdc 0x181e96eb4 0x181ea9d8c 0x19bfb69a0 0x1846defa8 0x18447322c 0x104a985b0 0x10b38c190)
libc++abi: terminating with uncaught exception of type NSException
* thread #1, queue = 'com.apple.main-thread', stop reason = signal SIGABRT
    frame #0: 0x00000001bb452a48 libsystem_kernel.dylib`__pthread_kill + 8
libsystem_kernel.dylib`__pthread_kill:
->  0x1bb452a48 <+8>:  b.lo   0x1bb452a64               ; <+36>
    0x1bb452a4c <+12>: stp    x29, x30, [sp, #-0x10]!
    0x1bb452a50 <+16>: mov    x29, sp
    0x1bb452a54 <+20>: bl     0x1bb44e6d0               ; cerror_nocancel
```

I have followed steps shown on this issue, with no luck: https://github.com/bharat-biradar/Google-Ml-Kit-plugin/issues/55.

```
    imageLabeler = GoogleMlKit.vision.imageLabeler(
      CustomImageLabelerOptions(
        customModel: CustomLocalModel.asset,
        customModelPath: "model.tflite",
      ),
    );
```

![image](https://user-images.githubusercontent.com/3192298/145568818-fbfc6ea2-74ec-41af-b250-f761e57a29c1.png)

Please let me know how to solve the below.

Thank you!

**Comment by yuexunshi** [comment]
me too.

**Comment by Holofox** [comment]
@dimitribocquet, @yuexunshi, you can use a workaround:

```dart
  Future<String> getModel(String assetPath) async {
    if (io.Platform.isIOS) {
      final path =
          '${(await getApplicationSupportDirectory()).path}/$assetPath';
      await io.Directory(dirname(path)).create(recursive: true);
      final file = io.File(path);

      if (!file.existsSync()) {
        final byteData = await rootBundle.load(assetPath);
        await file.writeAsBytes(byteData.buffer
            .asUint8List(byteData.offsetInBytes, byteData.lengthInBytes));
      }
      return file.path;
    }
    return 'flutter_assets/$assetPath';
  }
```

**pubspec.yaml**
```yaml
  assets:
    - assets/ml/
```

Sample:

```dart
final customModelPath = await getModel(
  'assets/ml/object_labeler.tflite',
);

_imageLabeler ??= GoogleMlKit.vision.imageLabeler(
  CustomImageLabelerOptions(
    customModel: CustomLocalModel.asset,
    customModelPath: customModelPath,
  ),
);
```

**Comment by yuexunshi** [comment]
> @dimitribocquet, @yuexunshi, you can use a workaround:
> 
> ```dart
> import 'dart:io' as io;
> import 'package:flutter/services.dart';
> import 'package:path/path.dart';
> import 'package:path_provider/path_provider.dart';
> 
> Future<String> getModel(String assetPath) async {
>   if (io.Platform.isAndroid) {
>     return 'flutter_assets/$assetPath';
>   }
>   final path = '${(await getApplicationSupportDirectory()).path}/$assetPath';
>   await io.Directory(dirname(path)).create(recursive: true);
>   final file = io.File(path);
> 
>   if (!await file.exists()) {
>     final byteData = await rootBundle.load(assetPath);
>     await file.writeAsBytes(byteData.buffer
>         .asUint8List(byteData.offsetInBytes, byteData.lengthInBytes));
>   }
>   return file.path;
> }
> ```
> 
> **pubspec.yaml**
> 
> ```yaml
>   assets:
>     - assets/ml/
> ```
> 
> Sample:
> 
> ```dart
> final customModelPath = await getModel(
>   'assets/ml/model.tflite',
> );
> 
> _imageLabeler ??= GoogleMlKit.vision.imageLabeler(
>   CustomImageLabelerOptions(
>     customModel: CustomLocalModel.asset,
>     customModelPath: customModelPath,
>   ),
> );
> ```

thank you。

**Comment by fbernaly** [comment]
Thanks @Holofox  for the workaround, we are including it in this PR: https://github.com/bharat-biradar/Google-Ml-Kit-plugin/pull/236

details here: https://github.com/bharat-biradar/Google-Ml-Kit-plugin/pull/236/commits/08f4e02d381688b3a0d0a23b8c9061d66611da71

it will be available in v0.2.0
